Sources/AblyAssetTrackingPublisher/DefaultPublisher.swift

```diff
@@ -527,21 +550,96 @@ extension DefaultPublisher {
         }
     }
     
-    private func saveLocationForFurtherSending(trackableId: String, location: EnhancedLocationUpdate) {
-        trackableState.skippedLocationsAdd(for: trackableId, location: location)
+    private func sendRawLocationUpdate(event: RawLocationChangedEvent, trackable: Trackable) {        
+        lastRawLocations[trackable] = event.locationUpdate.location
+        lastRawTimestamps[trackable] = event.locationUpdate.location.timestamp
+
+        event.locationUpdate.skippedLocations = rawLocationState.locationsList(for: trackable.id).map { $0.location }
+
+        rawLocationState.markMessageAsPending(for: trackable.id)
+
+        ablyPublisher.sendRawLocation(location: event.locationUpdate, trackable: trackable) { [weak self] result in
+            switch result {
+            case .failure(let error):
+                self?.enqueue(event: SendRawLocationFailureEvent(error: error, locationUpdate: event.locationUpdate, trackable: trackable))
+            case .success:
+                self?.enqueue(event: SendRawLocationSuccessEvent(trackable: trackable, location: event.locationUpdate.location))
+            }
+        }
+    }
+    
+    private func saveEnhancedLocationForFurtherSending(trackableId: String, location: EnhancedLocationUpdate) {
+        enhancedLocationState.addLocation(for: trackableId, location: location)
+    }
+    
+    private func saveRawLocationForFurtherSending(trackableId: String, location: RawLocationUpdate) {
+        rawLocationState.addLocation(for: trackableId, location: location)
+    }
+    
+    private func performRawLocationChanged(_ event: RawLocationChangedEvent) {
+        guard !state.isStoppingOrStopped, areRawLocationsEnabled == true else {
+            logger.error("Cannot perform EnhancedLocationChangedEvent. Publisher is not working.")
```

Comment:
The error message seems to be wrong :wink: 
```suggestion
            logger.error("Cannot perform RawLocationChangedEvent. Publisher is not working.")
```